nx/lib/nx/defn/evaluator.ex

```diff
@@ -305,7 +305,7 @@ defmodule Nx.Defn.Evaluator do
               "cannot pass a tensor expression as argument to defn, got: #{inspect(tensor)}"
 
       %Nx.Tensor{} = tensor ->
-        {tensor, caches}
+        {Nx.devectorize(tensor), caches}
```

Comment:
This fixes the test, but I don't think it's the correct solution, right?

Reply:
Ideally we wouldn't have anything vectorized here. If it is vectorized, it means it is leaking somewhere. But this is helpful. I will take a look!